view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js

```diff
@@ -366,16 +366,22 @@ define(
 
                 return '';
             },
+
             /**
-             * Get available card types translated to the Adyen card type codes
-             * (The card type alt code is the Adyen card type code)
+             * Fetches the brands array of the credit cards
              *
-             * @returns {string[]}
+             * @returns {array}
              */
-            // TODO check if we still need to rely on this or we can rely on the paymentMethods response types
-            getAvailableCardTypeAltCodes: function() {
-                var ccTypes = window.checkoutConfig.payment.ccform.availableTypesByAlt[this.getCode()];
-                return Object.keys(ccTypes);
+            getBrands: function() {
+                let paymentMethods =
+                    adyenPaymentService.getPaymentMethods()._latestValue.paymentMethodsResponse.paymentMethods;
+
+                for (let i = 0; i < paymentMethods.length; i++) {
+                    let paymentMethod = paymentMethods[i];
+                    if (Object.values(paymentMethod).includes("Credit Card")) {
```

Comment:
What do you think about relying on `"type": "scheme"` instead of `"name": "Credit Card"`. If a localisation is applied in the future to name parameter, `getBrands()` function may fail.